### DISCUSSION TITLE
Upload::make() confusion

### DISCUSSION THREAD

**Discussion by wsdnathan** [orchidsoftware/platform#1701] [Q&A] [answered]
For my screen: `Upload::make('model.myphotos')->groups('photo')->title('Upload Photos')`

The files upload as attachments and 'myphotos' gets set as ["11","12","13"] however when I return to the edit screen, the "Upload file" field is blank and not showing my uploaded files.

1. How do I get previously uploaded photos to display after returning to the edit screen?
2. What is the meaning of `->groups('photo')`?
3. Does 'myphotos' need to be a text or a json mysql field or does it not matter?
4. Are file uploads hash checked against existing files? I noticed they do not get written to filesystem again and instead the attachment points to older files uploaded via a difference crud resource.

**Comment by vallter2** [comment]
Try to add ->targetId() at the end...

**Reply by wsdnathan** [reply]
Just tried that but doesn't seem to have made any difference. The attachments exist in the table, the myphotos attribute gets reset with the new attachment IDs, however the Upload Photos field is still not showing any previously uploaded files after saving the record and returning to the edit screen (as if nothing was ever uploaded).

**Comment by wsdnathan** [comment, marked as answer]
The issue with the previously uploaded attachments failing to load seems related to this:

/vendor/orchid/platform/src/Screen/Fields/Upload.php line 116:
>         // set load relation attachment
>         $this->addBeforeRender(function () {
>            $value = Arr::wrap($this->get('value'));
>
>             if (! Assert::isIntArray($value)) {
>                 return;
>             }

From the database value, $value is being set to this after Arr::wrap():

> array:1 [▼
>   0 => "[78, 79, 80]"
> ]

This then always fails assertion `if (! Assert::isIntArray($value))`

If instead I set `$value = array(78,79,80);` then all the image attachments load as expected!

What am I missing here? Does the MySQL field attribute need to be of a specific type? Edit to note that I already tried JSON/text/varchar types with MySQL 8 and it results in the same failed assertion.

**Reply by tabuna** [reply]
Hi @wsdnathan. If you want to keep it in a column and not create a relationship. 
Then specify in the model castes - an array :

```php
namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Idea extends Model
{
    /**
     * The attributes that should be cast.
     *
     * @var array
     */
    protected $casts = [
        'files' => 'array',
    ];
}
```

Also, for reassurance, re-save the data in the model.

**Reply by wsdnathan** [reply]
@tabuna Thank you that works! I was hoping it would be something simple like that.

Curious, what is the proper way to create a relationship for this? I think this person is wondering the same thing: [https://github.com/orchidsoftware/platform/discussions/1709](url)

So to answer my original questions:

1. cast the field attribute as an array in the model
2. sets the attachments table group field
3. setting the photos/files field as a text type works
4. yes uploads seem to be hash checked, so new files will not be created if already exist and instead the new attachments record(s) will reference/point to the old file paths
